gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -104,34 +108,50 @@
    */
   @VisibleForTesting
   static final String KAFKA_TOPIC_SPECIFIC_STATE = "kafka.topic.specific.state";
+  public static final String KAFKA_SOURCE_WORK_UNITS_CREATION_THREADS = "kafka.source.work.units.creation.threads"; 
+  public static final int DEFAULT_THREAD_COUNT = 10;
```

Comment:
We process around 300 topics. I think we can set the thread size to 30. If we hit throttling issues from kafka or heap issues we can bring it down.